### PR TITLE
Intly 741

### PR DESCRIPTION
**Pull request by philbrookes**
## Verification
- Create a namespace, e.g. `keycloak`
- Deploy this operator to that namespace
- Create the keycloak resource
- Create the keycloak realm resource (without createOnly)
- Wait for everything to come up
- Add a role to a users client roles in the realm
- Check the role is added in the rhsso ui
- Remove a role from a users client roles in the realm
- Check the role is removed in the rhsso ui

### CODE REVIEW

File: deploy/examples/keycloakRealm.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
         "realm": "arealm",
         "displayName": "arealm",
         "enabled": true,
-        "eventsListeners": ["metrics-listener"],
```
Comment (pb82) [inline deploy/examples/keycloakRealm.json]: why remove the metrics listener from the example?

Comment (philbrookes) [reply deploy/examples/keycloakRealm.json]: oh, it kept generating an error that was a bit distracting, I can re-add it though

Comment (philbrookes) [reply deploy/examples/keycloakRealm.json]: done.

File: pkg/keycloak/realm/phaseHandler.go [hunk 2]
```diff
@@ -247,6 +246,77 @@ func (ph *phaseHandler) reconcileUser(kcUser, specUser *v1alpha1.KeycloakUser, r
...
+	if err != nil {
+		return err
+	}
+	for clientName, roles := range specUser.ClientRoles {
+		rolesCopy := make([]string, len(roles))
+		copy(rolesCopy, roles)
+	FindMatchingClient:
+		for _, client := range clients {
+			if clientName == client.ClientID {
+				err = ph.reconcileRolesForClient(rolesCopy, client, specUser, realmName, authenticatedClient)
```
Comment (maleck13) [inline pkg/keycloak/realm/phaseHandler.go]: ignoring err here we should at least log it and perhaps gather a multi error?

Comment (philbrookes) [reply pkg/keycloak/realm/phaseHandler.go]: good spot, fixed now

**Comment by philbrookes** [comment]
@maleck13 @pb82 any further thoughts on this PR?

**Comment by pb82** [comment]
@philbrookes will verify it shortly

**Pull request merged**